dockerfiles/Dockerfile.sandbox

```diff
@@ -18,9 +18,14 @@ FROM tiangolo/uwsgi-nginx-flask:python3.10
 # Install dependencies required for Lean 4, pypy3, and other tools
 RUN apt-get update && \
     apt-get install -y curl git net-tools bzip2 build-essential libseccomp-dev && \
-    curl -L https://downloads.python.org/pypy/pypy3.10-v7.3.17-linux64.tar.bz2 -o /tmp/pypy.tar.bz2 && \
+    case "$TARGETARCH" in \
+        amd64) PYPY_ARCH=linux64 ;; \
+        arm64|aarch64) PYPY_ARCH=aarch64 ;; \
+    *) echo "Unsupported TARGETARCH '$TARGETARCH'" >&2; exit 1 ;; \
+    esac && \
+    curl -L https://downloads.python.org/pypy/pypy3.10-v7.3.17-$PYPY_ARCH.tar.bz2 -o /tmp/pypy.tar.bz2 && \
     tar -xjf /tmp/pypy.tar.bz2 -C /opt/ && \
-    ln -s /opt/pypy3.10-v7.3.17-linux64/bin/pypy3 /usr/bin/pypy3 && \
+    ln -s /opt/pypy3.10-v7.3.17-$PYPY_ARCH/bin/pypy3 /usr/bin/pypy3 && \
     /usr/bin/pypy3 -m ensurepip && \
     rm /tmp/pypy.tar.bz2
```

Comment:
Always a good practice to use for all apt-related layers:
```shell
rm -rf /var/cache/apt/archives /var/lib/apt/lists/*
```

See https://docs.docker.com/build/building/best-practices/#apt-get.

Reply:
@activatedgeek where should I add this line?